### ISSUE TITLE
Incorrect variables in write_restart()

### ISSUE THREAD

**Issue by tclune** [GEOS-ESM/MAPL#3013]
The procedure shown below is attempting to write (uninitialized) local variables states rather than the states provided as dummy arguments.   (Note that I'm using ESMF camelCase for common ESMF variables rather than the MAPL usual snake_case.)

https://github.com/GEOS-ESM/MAPL/blob/532f6c98f2b55c0865ecd95888b6d1c5332393a7/generic3g/OuterMetaComponent/write_restart.F90#L38-L39

**Comment by tclune** [comment]
Same issue with `read_restart()`

Found these while trying to cleanup various compiler warnings.

**Comment by pchakraborty** [comment]
> The procedure shown below is attempting to write (uninitialized) local variables states rather than the states provided as dummy arguments. (Note that I'm using ESMF camelCase for common ESMF variables rather than the MAPL usual snake_case.)
> 
> https://github.com/GEOS-ESM/MAPL/blob/532f6c98f2b55c0865ecd95888b6d1c5332393a7/generic3g/OuterMetaComponent/write_restart.F90#L38-L39

The local variables `import/internal_state` are getting filled in the previous steps
```Fortran
   35      call states%get_state(import_state, "import", _RC)
   36      call states%get_state(internal_state, "internal", _RC)
```
For the internal state, that seems to be right way. For import state, I'm not sure why I didn't use `importState` input. Maybe for consistency?

**Comment by tclune** [comment]
@pchakraborty Same question as above:  Can this issue be closed?

**Comment by pchakraborty** [comment]
Yes, this is not a bug. The local variables get filled in a previous step.

Closing the issue.
